File: Exercise/Hard/14. Funny Numbers/Code/Funny Numbers.ts
function FunnyNumber(n: number, p: number): number {
    const digits = n.toString().split('').map(Number);
    const sum = digits.reduce((acc, digit, i) => acc + Math.pow(digit, p + i), 0);
    return sum % n === 0 ? sum / n : -1;
}

console.log(FunnyNumber(89, 1));    
console.log(FunnyNumber(695, 2));   
console.log(FunnyNumber(46288, 3));
console.log(FunnyNumber(92, 1));    

File: Exercise/Hard/14. Funny Numbers/Code/Funny Numbers.cs
using System;

class Program {
    static int FunnyNumber(int n, int p) {
        int sum = 0;
        string str = n.ToString();
        
        for (int i = 0; i < str.Length; i++) {
            int digit = str[i] - '0';
            sum += (int)Math.Pow(digit, p + i);
        }

        if (sum % n == 0) {
            return sum / n;
        } else {
            return -1;
        }
    }

    static void Main() {
        int n = int.Parse(Console.ReadLine());

        int p = int.Parse(Console.ReadLine());
        
        Console.WriteLine(FunnyNumber(n, p));
    }
}



